configs/configschema/schema.go

```diff
@@ -21,6 +28,11 @@ type Block struct {
 	// BlockTypes describes any nested block types that may appear directly
 	// inside the block.
 	BlockTypes map[string]*NestedBlock
+
+	Description     string
+	DescriptionKind StringKind
+
+	Deprecated bool
```

Comment:
The configschema is intended to describe the structure , rather than how the content is used. Adding fields has already proved troublesome, for example we can't really use `MinItems` and `MaxItems` on nested blocks. I'd rather sort out necessary schema changes separately if they're not required for provider-metas.

Reply:
/cc @paultyng

Reply:
I believe this is the only way to get them to the JSON schema dump, so the only alternative would be to plumb that path for schema information differently which may be undesirable.